autogluon/task/tabular_prediction/predictor.py

```diff
@@ -315,6 +315,42 @@ def fit_summary(self, verbosity=3):
             print("*** End of fit() summary ***")
         return results
 
+    # TODO: Consider adding time_limit option to early stop the feature importance process
+    def feature_importance(self, model=None, dataset=None, features=None, raw=True, silent=False):
+        """
+        Calculates feature importance scores for the given model.
+        Feature scores represent the score reduction the model achieves when the provided data's feature is randomly shuffled among its rows.
+        A score of 0.01 would indicate that the score dropped by 0.01 when the feature was randomly shuffled.
+        The higher the score a feature has, the more important it is to the model's performance.
+        If a feature has a negative score, this means that the feature is likely harmful to the final model, and a model trained with the feature removed would be expected to achieve a better score.
+        Note that calculating feature importance can be a very computationally expensive process, particularly if the model uses hundreds or thousands of features. In many cases, this can take longer than the original model training.
+        To estimate how long `feature_importance(model, dataset, features)` will take, it is roughly the time taken by `predict_proba(dataset, model)` multiplied by the number of features.
+
+        Parameters
+        ----------
+        model : str, default = None
+            Model to get feature importances for, if None the best model is chosen.
+        dataset : str or :class:`TabularDataset` or `pandas.DataFrame` (optional)
+            This Dataset must also contain the label-column with the same column-name as specified during fit().
+            If specified, then the dataset is used to calculate the feature importance scores.
+            If str is passed, `dataset` will be loaded using the str value as the file path.
+            If not specified, the original dataset used during fit() will be used if `cache_data=True`. Otherwise, an exception will be raised.
+            Do not pass the training data through this argument, as the feature importance scores calculated will be inaccurate.
+        features : list, default = None
+            List of str feature names that feature importances are calculated for and returned, specify None to get all feature importances.
+        raw : bool, default = True
+            Whether to compute feature importance on raw original features (after automated feature engineering) or on the features used by the particular model.
```

Comment:
consider changing: 

"raw original features"
-> 
"raw original data features"

I'd also add more explanation: 
"or on the features used by the particular model (for example, a stacker model uses both original data features as well as predictions from lower-level models)"

Reply:
Updated